packages/connection-form/src/components/advanced-options-tabs/tls-ssl-tab/tls-certificate-authority.tsx

```diff
@@ -26,20 +38,38 @@ function TLSCertificateAuthority({
     <FormFieldContainer className={caFieldsContainer}>
       <FileInput
         description={'Learn More'}
-        disabled={disabled}
+        disabled={disabled || useSystemCA}
         id="tlsCAFile"
         dataTestId="tlsCAFile-input"
         label="Certificate Authority (.pem)"
         link={
           'https://docs.mongodb.com/manual/reference/connection-string/#mongodb-urioption-urioption.tlsCAFile'
         }
         onChange={(files: string[] | null) => {
-          updateCAFile(files && files.length > 0 ? files[0] : null);
+          updateCAFile(files && files.length > 0 ? files[0] : null, false);
         }}
         showFileOnNewLine
         values={caFile ? [caFile] : undefined}
         optional
       />
+      <Checkbox
+        onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+          updateCAFile(null, event.target.checked);
+        }}
+        id="useSystemCA-input"
+        data-testid="useSystemCA-input"
+        label="Use System Certificate Authority"
+        disabled={disabled}
+        checked={useSystemCA}
+        bold={false}
+      />
+      <Description
```

Comment:
@Anemy found out that both label and description should be passed as "Label" for checkboxes, should be already changed on main for the rest of the connection form.

Reply:
Ah thanks for pointing that out, I missed that – updated to match those changes :)